client/src/commands/associateIdentityWithNode.ts

```diff
@@ -0,0 +1,188 @@
+            if (!chosenNode) {
+                return;
+            }
+
+            node = chosenNode.data;
+        }
+
+        let walletRegistryEntry: FabricWalletRegistryEntry;
+        let walletMessage: string;
+
+        if (node.type === FabricNodeType.PEER) {
+            walletMessage = `An admin identity for "${node.msp_id}" is required to perform installs. Which wallet is it in?`;
+        } else if (node.type === FabricNodeType.ORDERER) {
+            walletMessage = `An admin identity for "${node.msp_id}" is required to perform instantiates. Which wallet is it in?`;
+        } else if (node.type === FabricNodeType.CERTIFICATE_AUTHORITY && !node.enroll_id && !node.enroll_secret) {
+            walletMessage = `An Admin identity for ${node.name} is required to register identities. Which wallet is it in?`;
+        }
+
+        let enroll: boolean = false;
+        if (node.type === FabricNodeType.CERTIFICATE_AUTHORITY && node.enroll_id && node.enroll_secret) {
+            const enrollNew: string = 'Use ID and secret to enroll a new identity';
+            const chooseExisting: string = 'Choose an existing identity';
+            const chosenMethod: string = await UserInputUtil.showQuickPick('The json for this certificate authority includes an enrollment ID and secret...', [enrollNew, chooseExisting]);
+            if (!chosenMethod) {
+                return;
+            }
+
+            if (chosenMethod === enrollNew) {
+                walletMessage = 'Which wallet do you want to add the admin identitiy to?';
+                enroll = true;
+            } else {
+                walletMessage = 'Which wallet is the admin identity in?';
+            }
+        }
+
+        const chosenWallet: IBlockchainQuickPickItem<FabricWalletRegistryEntry> = await UserInputUtil.showWalletsQuickPickBox(walletMessage, false, true);
+        if (!chosenWallet) {
+            return;
+        }
+
+        if (!chosenWallet.data) {
+            walletRegistryEntry = await vscode.commands.executeCommand(ExtensionCommands.ADD_WALLET) as FabricWalletRegistryEntry;
+            if (!walletRegistryEntry) {
+                return;
+            }
+        } else {
+            walletRegistryEntry = chosenWallet.data;
+        }
+
+        node.wallet = walletRegistryEntry.name;
+
+        const walletGenerator: IFabricWalletGenerator = FabricWalletGeneratorFactory.createFabricWalletGenerator();
+        const wallet: IFabricWallet = walletGenerator.getNewWallet(walletRegistryEntry.walletPath);
+
+        if (enroll) {
+            node = await enrollIdAndSecret(node, wallet);
+            if (!node) {
+                return;
+            }
+        } else {
+            const identitiies: Array<string> = await wallet.getIdentityNames();
+            let identityMessage: string = `Select the admin identity for ${node.msp_id}`;
```

Comment:
It's probably worth checking `if(!node.msp_id){` before we assign this (as if it's undefined, we'll need to change it anyway).

Reply:
in the majority of cases it will be set so might as well set it then change it if need to